### PR TITLE
Avoid `UnboundLocalError` or duplicates in results

### PR DESCRIPTION
**Pull request by gbastien**
when using `@search` endpoint and a brain is orphan or a `KeyError` occurs during result serialization

See issue #1231

PS: if this is backported to version working with Zope2, there will need to change as getObject an orphan brain raises KeyError in Zope4 but an AttributeError in Zope2...

### CODE REVIEW

**Comment by gbastien** [comment]
Hi @sneridagh @tisto 

when you have time to look at this?  We overrided in our code so that is not urgent, but I think this could happen to other people as well.

Thank you for great job and support!

Gauthier

**Review by tisto — APPROVED** [review]
@gbastien thanks for your fix and PR! LGTM! I'd like to get a second review, then I can merge and release it right away.

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**
